snorkel/contrib/fonduer/fonduer/parser.py

```diff
@@ -247,18 +252,22 @@ def parse_node(node, table_info=None):
 
             for child in node:
                 if child.tag == 'table':
-                    parse_node(child, TableInfo(document=table_info.document))
+                    parse_node(child, TableInfo(document=table_info.document), figure_info)
+                elif child.tag == '':
+                    parse_node(child, table_info, FigureInfo(document=figure_info.document))
```

Comment:
Checking if a tag is empty to imply that we need a FigureInfo seems a little confusing, is a Figure really the only scenario where the tag is empty?